logstash-core/lib/logstash/runner.rb

```diff
@@ -501,4 +512,15 @@ def setting(key)
     @settings.get_value(key)
   end
 
+  def verify_deprecated_setting(old_setting, new_setting)
+    if @settings.set?(old_setting)
+      logger.warn("Deprecated setting `#{old_setting}` please use `#{new_setting}`")
+      if @settings.set?(new_setting)
```

Comment:
I would consider both being set to be an error condition, and have a preference for failure when given conflicting directives.